controllers/users.js

```diff
@@ -738,11 +738,34 @@ const filterUsers = async (req, res) => {
       return res.boom.badRequest("filter for item not provided");
     }
 
-    const users = await dataAccess.retreiveFilteredUsers(req.query);
+    const { page, limit } = req.query;
+    const pageNumber = parseInt(page) || 0;
+    const limitNumber = parseInt(limit) || 100;
+    const skip = (pageNumber - 1) * limitNumber;
+
+    const users = await dataAccess.retreiveFilteredUsers(req.query, skip, limitNumber);
+    const totalCount = await users.length;
+    const totalPages = Math.ceil(totalCount / limitNumber);
+    const nextPage = pageNumber < totalPages - 1 ? pageNumber + 1 : null;
+    const prevPage = pageNumber > 0 ? pageNumber - 1 : null;
+
+    const paginationLinks = {
+      next: nextPage ? getFilteredPaginationLink(req.query, nextPage, limitNumber) : null,
+      prev: prevPage ? getFilteredPaginationLink(req.query, prevPage, limitNumber) : null,
+    };
+
     return res.json({
       message: users.length ? "Users found successfully!" : "No users found",
       users: users,
       count: users.length,
+      pagination: {
+        totalCount,
+        totalPages,
+        currentPage: pageNumber,
+        nextPage,
+        prevPage,
+        links: paginationLinks,
+      },
```

Comment:
let's keep our pagination format the same throughout, please have a look at others API's where pagination is done

Reply:
cool, will look into it and I will try my level best to do it same way.

Reply:
@prakashchoudhary07 changed it. Can you please look into it?